service/src/main/java/bio/terra/workspace/service/resource/controlled/flight/clone/bucket/BucketCloneRolesService.java

```diff
@@ -80,14 +84,23 @@ private void addOrRemoveBucketIdentities(
         storageCow.getIamPolicy(inputs.getBucketName()).toBuilder();
     for (String roleName : inputs.getRoleNames()) {
       switch (operation) {
-        case ADD:
-          policyBuilder.addIdentity(Role.of(roleName), saIdentity);
-          break;
-        case REMOVE:
-          policyBuilder.removeIdentity(Role.of(roleName), saIdentity);
-          break;
+        case ADD -> policyBuilder.addIdentity(Role.of(roleName), saIdentity);
+        case REMOVE -> policyBuilder.removeIdentity(Role.of(roleName), saIdentity);
       }
     }
-    storageCow.setIamPolicy(inputs.getBucketName(), policyBuilder.build());
+    final Policy newPolicy = policyBuilder.build();
+    storageCow.setIamPolicy(inputs.getBucketName(), newPolicy);
+
+    // verify the role changes have propagated, as we may need to work on this bucket immediately
```

Comment:
optional - I see there's GcpUtils.pollUntilSuccess. You could create a similar GcpUtils.pollUntilSuccess that takes in a function instead of OperationCow.